From: Sebastian Keller <skeller@gnome.org>
Date: Sat, 24 Jan 2026 18:15:43 +0100
Subject: appDisplay: Only create default folders on the first start

If all folders were removed, after a session restart the default folders
would have been recreated again, effectively preventing users from
using the app grid without folders.

Avoid this problem by only creating the default folders if the
folder-children setting has never been modified before.

Related: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/8713
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4047>
(cherry picked from commit 889c1a444fcfd0588716528b8ee7dd858a610694)
Origin: upstream, 49.4, commit:b0d396c8238b15a30b3a3321bbb257a913d6c63f
---
 js/ui/appDisplay.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index b473848..71cc36a 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -1403,7 +1403,7 @@ class AppDisplay extends BaseAppView {
     }
 
     _ensureDefaultFolders() {
-        if (this._folderSettings.get_strv('folder-children').length > 0)
+        if (this._folderSettings.get_user_value('folder-children') !== null)
             return;
 
         const appSys = Shell.AppSystem.get_default();
